### ISSUE TITLE
False positive Use of moved value

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#6826]
## Environment

* **IntelliJ Rust plugin version:** 0.3.142.3661-203-nightly
* **Rust toolchain version:** 1.48.0 (7eac88abb 2020-11-16) x86_64-apple-darwin
* **IDE name and version:** IntelliJ IDEA 2020.3.2 Ultimate Edition (IU-203.7148.57)
* **Operating system:** macOS 10.15.6
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

![image](https://user-images.githubusercontent.com/2539310/107560797-dc755e80-6bee-11eb-828f-8fedb0fa2cf9.png)

## Steps to reproduce
`src/main.rs`
```rust
use serde_json::Value;
use std::convert::TryFrom;
use std::borrow::Cow;
use serde::{Serialize,Deserialize};

#[derive(Debug,Clone,Serialize,Deserialize,Eq,PartialEq)]
struct OcppCall {
    message_id: String,
    action: String,
    payload: Value,
}

impl TryFrom<Value> for OcppCall {
    type Error = ();

    fn try_from(value: Value) -> Result<Self, ()> {
        let mut items : [Value; 4] = serde_json::from_value(value.clone()).unwrap();

        if let [
        Value::Number(n),
        Value::String(message_id), // warning
        Value::String(action), // warning
        payload @ Value::Object(_)
        ] = items {
            Ok(Self {
                message_id,
                action,
                payload
            })
        } else {
            Err(())
        }
    }
}

fn main() {}
```

`Cargo.toml`
```toml

[package]
name = "rust_project"
version = "0.1.0"
authors = []
edition = "2018"


[dependencies]
serde = { version = "1.0.123", features = ["derive"] }
serde_json = "1.0.62"
```

**Comment by seeekr** [comment]
This was fine up until a few days ago, when, I assume, a new Intellij Rust plugin version was downloaded (?), and now this error shows up all over my code, and I'd have to add way too many .clone()-s on primitive value bindings to fix this manually... :)

**Comment by seeekr** [comment]
Never mind, it's not fixed on nightly plugin. Seems like the IDE just hadn't done its parsing + checking yet when I tested it, not sure why, but now it does so and the false positive error highlightings are back, e.g.:
<img width="453" alt="CleanShot 2023-05-16 at 22 38 35@2x" src="https://github.com/intellij-rust/intellij-rust/assets/302886/a8b4c2c5-fced-4a7b-a4c9-f1f3bbbe9cfd">
(`zstdlevel` is Copy, and the compiler is fine with the code.)

**Comment by cgudrian** [comment]
It's even complaining when using plain `u16` variables:
![image](https://github.com/intellij-rust/intellij-rust/assets/204897/a56b776f-e5e3-48b3-b2d6-7d7b388653ad)

I'm on Rust nightly-2023-05-18.

**Comment by stevepryde** [comment]
This is still showing up all over my code in the latest plugin. I know it's moved to youtrack now but find that interface confusing and unusable

**Comment by seeekr** [comment]
In my codebase, this primarily happens when the code in question is inside of a tokio::select! macro invocation body.
(I, too, find YouTrack unpleasant to use.)
